pkg/controllers/node/node_controller.go

```diff
@@ -79,6 +80,11 @@ func NewNodeController(ctx context.Context, k8sClientset *kubernetes.Clientset,
 		nodeCache:    make(map[string]*api.Node),
 	}
 
+	// Create a default profile for auto hostendpoints
+	err := nc.createAllowProfile()
```

Comment:
This processing is very imperative.

We want our controllers to work declarative.  We have some defined state that we want the system to be in (in this case, the allow profile exists), then we have a control loop that:

1. Reads the state
2. Compares with our desired state
3. If different, makes a change

(point 1 might be via a watch, rather than getting in a loop)